gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/JobStatistics.java

```diff
@@ -325,13 +338,26 @@ public Long totalBytesProcessed() {
       return totalBytesProcessed;
     }
 
+    /**
+     * Returns the query plan as a list of stages. Each stage involves a number of steps that read
+     * from data sources, perform a series of transformations on the input, and emit an output to a
+     * future stage (or the final result). The query plan is available for a completed query job and
+     * is retained for 7 days.
+     *
+     * @see <a href="https://cloud.google.com/bigquery/query-plan-explanation">Query Plan</a>
+     */
+    public List<QueryStage> queryPlan() {
+      return queryPlan;
```

Comment:
Oh, that sounds strange to me. Indeed for dry run not all values may be available (especially time related - if can't be estimated) however I thought that could be helpful to estimate how expensive a query might be and how it would be performed (similar to the SQL explain plan which does not really run the query, hence "dry run").

/cc @tigani